### ISSUE TITLE
JIT: assert seen when running tests via run-xunit-perf.cmd

### ISSUE THREAD

**Issue by AndyAyersMS** [dotnet/coreclr#7954]
To repro:

- build x64 checked

- cd tests && runtest x64 checked, or at least runtest x64 checked GenerateLayoutOnly

- cd <root> && tests\scripts\run-xunit-perf.cmd -arch x64 -configuration Checked -testBinLoc bin\tests\Windows_NT.x64.Checked\JIT\Performance\CodeQuality\Roslyn

Result is:
```
--- setting up sandbox
--- Running CscBench

Assert failure(PID 22384 [0x00005770], Thread: 15688 [0x3d48]): Assertion failed 'lvaTable[lhsLclNum].lvType != TYP_STRUCT' in '<InvokeTestAsync>d__4:MoveNext():this' (IL size 333)

    File: c:\repos\coreclr\src\jit\liveness.cpp Line: 79
    Image: c:\repos\coreclr\sandbox\corerun.exe
```

This works at 09c9b00 and fails at the subsequent 9c95461

**Comment by CarolEidt** [comment]
Thanks for identifying this. The problem is that liveness doesn't expect a struct assignment with the same lclVar on the lhs and rhs. I'm working on a fix that eliminates the assert, but also removes an assignment from a struct to itself, because I believe that may cause problems if a helper call is generated.

**Comment by CarolEidt** [comment]
This fix for this was incorporated into the (fixed) StructOpts changes #8057
